core/src/main/java/com/epam/esm/dao/CertificateJdbcDao.java

```diff
@@ -0,0 +1,154 @@
+                    " last_update_date) values(?, ?, ?, ?, ?, ?)";
+    private static final String GET_BY_TAG_NAME =
+            "SELECT * FROM gift_certificate JOIN (" +
+                    " SELECT tag.name as tag_name, tag.id as tag_id, certificate_id FROM certificate_tag" +
+                    " JOIN tag ON (tag_id = tag.id) WHERE tag.name = ?) " +
+                    " AS new_tag ON (gift_certificate.id = new_tag.certificate_id)" +
+                    " ORDER BY gift_certificate.id";
+    private static final String GET_ALL =
+            "SELECT * FROM gift_certificate LEFT JOIN (" +
+                    " SELECT tag.name as tag_name, tag.id as tag_id, certificate_id FROM certificate_tag" +
+                    " JOIN tag ON (tag_id = tag.id)) AS new_tag ON (gift_certificate.id = new_tag.certificate_id)" +
+                    " ORDER BY gift_certificate.id";
+
+    private static final String GET_BY_NAME_PART =
+            "SELECT * FROM gift_certificate JOIN (" +
+                    " SELECT tag.name as tag_name, tag.id as tag_id, certificate_id FROM certificate_tag" +
+                    " JOIN tag ON (tag_id = tag.id)) AS new_tag ON (gift_certificate.id = new_tag.certificate_id)" +
+                    " WHERE name LIKE ? ORDER BY gift_certificate.id";
+    private static final String DELETE_BY_ID =
+            "DELETE FROM gift_certificate WHERE id = ?";
+    private static final String FIND_CERTIFICATE_TAG =
+            "SELECT COUNT(*) FROM certificate_tag WHERE certificate_id = ? AND tag_id = ?";
+
+    private static final String ID = "ID";
+    private final CertificateResultSetExtractor certificateResultSetExtractor;
+    private final JdbcTemplate jdbcTemplate;
+
+    @Autowired
+    public CertificateJdbcDao(CertificateResultSetExtractor certificateResultSetExtractor, JdbcTemplate jdbcTemplate) {
+        this.certificateResultSetExtractor = certificateResultSetExtractor;
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public List<Certificate> getAll() {
+        return jdbcTemplate.query(GET_ALL, certificateResultSetExtractor);
+    }
+
+    @Override
+    public List<Certificate> getByTagName(String tagName) {
+        return jdbcTemplate.query(GET_BY_TAG_NAME, certificateResultSetExtractor, tagName);
+    }
+
+    @Override
+    public List<Certificate> getByNamePart(String namePart) {
+        String formattedNamePart = "%" + namePart + "%";
+        return jdbcTemplate.query(GET_BY_NAME_PART, certificateResultSetExtractor, formattedNamePart);
+    }
+
+    @Override
+    public Optional<Certificate> getById(long id) {
+        List<Certificate> query = jdbcTemplate.query(GET_BY_ID, certificateResultSetExtractor, id);
+        if (query.size() > 1) {
+            throw new MultipleRecordsWereFoundException(MULTIPLE_RECORDS_WERE_FOUND_BY_ID, id);
+        }
+        return query.stream().findAny();
+    }
+
+    @Override
+    public void update(Certificate entity) {
```

Comment:
@KukharchukDaniil Could you return updated entity? It helps check all changes.

Reply:
fixed